pkg/config/config.go

```diff
@@ -18,49 +18,47 @@
 package config
 
 import (
-	"io/ioutil"
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"os"
+	"path/filepath"
 
 	"github.com/BurntSushi/toml"
 )
 
 const (
 	configDirectory = "./res"
-	configDefault = "configuration.toml"
-	configDocker = "configuration-docker.toml"
-	configUnitTest = "configuration-test.toml"
+	configDefault   = "configuration.toml"
+
+	configDirEnv = "EDGEX_CONF_DIR"
 )
 
 var confDir = flag.String("confdir", "", "Specify local configuration directory")
 
 func LoadFromFile(profile string, configuration interface{}) error {
 	path := determinePath()
-	fileName := determineConfigFile(profile)
-	contents, err := ioutil.ReadFile(path + "/" + fileName)
+	fileName := path + "/" + determineConfigFile(profile)
+
+	contents, err := ioutil.ReadFile(fileName)
 	if err != nil {
-		return fmt.Errorf("could not load configuration file (%s): %v", path, err.Error())
+		return fmt.Errorf("could not load configuration file (%s): %v", fileName, err.Error())
 	}
 
 	// Decode the configuration from TOML
 	err = toml.Unmarshal(contents, configuration)
 	if err != nil {
-		return fmt.Errorf("unable to parse configuration file (%s): %v", path, err.Error())
+		return fmt.Errorf("unable to parse configuration file (%s): %v", fileName, err.Error())
 	}
 
 	return nil
 }
 
 func determineConfigFile(profile string) string {
-	switch profile {
-	case "docker":
-		return configDocker
-	case "unit-test":
-		return configUnitTest
-	default:
+	if profile == "" {
```

Comment:
This causes a bug. By default, the value of profile is "default" -- thus the switch case from the previous version. As this code is written, when the services start the following error is thrown by "make run":

ERROR: 2018/04/26 10:33:47 could not load configuration file (./res/configuration-default.toml): open ./res/configuration-default.toml: no such file or directory

Reply:
I can think of two different solutions:
 - test also for the string "default"
 - we can set the default profile to the empty string.

I prefer the second option as there is no 'magic' strings, just the default go value for a string. But I'm ok with both.

Reply:
Either is fine. In doing the former, you don't have to change the flags logic in the main.go of all the services. If the latter, then you do have to change it. It depends on your preference for limiting the scope of change in this PR.

Reply:
Ah, I see you already decided to do the latter ;-)

Reply:
Yes, I like it best. But I can change to to "default" if you think it is better.